### PR TITLE
Test for PVC creation and deletion from UI

### PR DESCRIPTION
**Pull request by am-agrawa**

### CODE REVIEW

File: ocs_ci/ocs/ui/base_ui.py [hunk 1]
```diff
@@ -402,7 +402,7 @@ def login_ui():
 
         # headless browsers are web browsers without a GUI
         headless = ocsci_config.UI_SELENIUM.get("headless")
-        if headless:
+        if not headless:
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/base_ui.py:405]: why did you change it?
jenkins slave works without console.

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 2]
```diff
@@ -51,9 +51,14 @@ def create_pvc_ui(self, sc_type, pvc_name, access_mode, pvc_size):
         logger.info("Select PVC size")
         self.do_send_keys(self.pvc_loc["pvc_size"], text=pvc_size)
 
+        logger.info(f"Select Volume Mode {vol_mode}")
```
Comment (ebondare) [inline ocs_ci/ocs/ui/pvc_ui.py:54]: I think we should check OCP version here and choose the volume mode only for 4.8 version or later.

File: ocs_ci/ocs/ui/views.py [hunk 3]
```diff
@@ -112,7 +116,7 @@
     "ReadOnlyMany": ('input[data-test="Read Only (ROX)-radio-input"]', By.CSS_SELECTOR),
     "pvc_size": ('input[data-test="pvc-size"]', By.CSS_SELECTOR),
     "pvc_create": ('button[data-test="create-pvc"]', By.CSS_SELECTOR),
-    "pvc_actions": ('button[data-test-id="actions-menu-button"]', By.CSS_SELECTOR),
+
```
Comment (ebondare) [inline ocs_ci/ocs/ui/views.py:119]: If "button[aria-label='Actions']" works for 4.6 and 4.7, I think it would be better to just substitute it here. If not, you could add it to 4.8 dictionary without removing it in the 4.6/4.7 one. We might want to run regression tests on later versions of 4.7 so it's better not to break the existing tests.

File: tests/ui/test_pvc_ui.py [hunk 4]
```diff
@@ -23,22 +23,37 @@ def teardown(self):
...
+                *["ocs-storagecluster-ceph-rbd", "test-pvc-rbd", "ReadWriteMany", "3", "Filesystem"]
+            ),
+            pytest.param(
+                *["ocs-storagecluster-ceph-rbd-thick", "test-pvc-rbd-thick", "ReadWriteMany", "4", "Filesystem"]
+            ),
+            pytest.param(
+                *["ocs-storagecluster-cephfs", "test-pvc-fs", "ReadWriteOnce", "10", None]
+            ),
+            pytest.param(
+                *["ocs-storagecluster-ceph-rbd", "test-pvc-rbd", "ReadWriteOnce", "11", "Block"]
```
Comment (ebondare) [inline tests/ui/test_pvc_ui.py:41]: I think it would be ideal to have both Block and Filesystem rbd PVCs with ReadWriteOnce and check that the volume mode of the PVC that gets created really matches the chosen volume mode.

**Review by ebondare — CHANGES_REQUESTED** [review]

File: tests/ui/test_pvc_ui.py [hunk 5]
```diff
@@ -23,22 +23,37 @@ def teardown(self):
     @ui
```
Comment (ebenahar) [inline tests/ui/test_pvc_ui.py:23]: since we don't have jobs for the ui tier, these tests should be part of tierx

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 6]
```diff
@@ -19,7 +20,7 @@ def __init__(self, driver):
         ocp_version = get_ocp_version()
         self.pvc_loc = locators[ocp_version]["pvc"]
 
-    def create_pvc_ui(self, sc_type, pvc_name, access_mode, pvc_size):
+    def create_pvc_ui(self, sc_type, pvc_name, access_mode, pvc_size, vol_mode):
```
Comment (prsurve) [inline ocs_ci/ocs/ui/pvc_ui.py:23]: doc string missing

File: ocs_ci/ocs/ui/base_ui.py (hunk 1 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py]: change reverted

File: requirements.txt [hunk 7]
```diff
@@ -1,2 +1,30 @@
...
 -e .
+
+pytest~=5.3.5
+PyYAML~=5.4.1
+botocore~=1.17.63
+boto3~=1.14.7
+numpy~=1.19.5
+elasticsearch~=7.8.0
+selenium~=3.141.0
+prettytable~=0.7.2
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply requirements.txt]: change reverted

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 2 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: checking OCP version now for volume mode

File: ocs_ci/ocs/ui/views.py (hunk 3 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/views.py]: added

File: tests/ui/test_pvc_ui.py [hunk 8]
```diff
@@ -20,25 +21,43 @@ def teardown(self):
         pvcs = [pvc_obj for pvc_obj in pvc_objs if "test-pvc" in pvc_obj.name]
         delete_pvcs(pvc_objs=pvcs)
 
-    @ui
+    @tier1
```
Comment (prsurve) [inline tests/ui/test_pvc_ui.py:24]: are we planning to run this as tier1 because there is a marker change

File: tests/ui/test_pvc_ui.py (hunk 4 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py:42]: added

File: tests/ui/test_pvc_ui.py (hunk 5 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: changed, please check

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 9]
```diff
@@ -80,3 +88,7 @@ def delete_pvc_ui(self, pvc_name):
 
         logger.info("Confirm PVC Deletion")
         self.do_click(self.pvc_loc["confirm_pvc_deletion"])
+
+        time.sleep(2)
+
+
```
Comment (prsurve) [inline ocs_ci/ocs/ui/pvc_ui.py:94]: Remove extra line

**Review by prsurve — COMMENTED** [review]
- Resolve Conflicts
- You need to signed off the code

File: tests/ui/test_pvc_ui.py (hunk 8 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py:24]: yes, it was discussed in the last meeting. @ebenahar @OdedViner  Please confirm.

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 6 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py:23]: added

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 9 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: removed

File: tests/ui/test_pvc_ui.py (hunk 8 above)
Comment (OdedViner) [reply tests/ui/test_pvc_ui.py:24]: yes

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 10]
```diff
@@ -70,7 +81,7 @@ def delete_pvc_ui(self, pvc_name):
         self.do_send_keys(self.pvc_loc["search_pvc"], text=pvc_name)
 
         logger.info(f"Go to PVC {pvc_name} Page")
-        self.do_click(self.pvc_loc["pvc_test"])
+        self.do_click(pvc_4_8b[pvc_name])
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/pvc_ui.py:84]: is it possible to pass a generic dict and not one that is for a specific version?

Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: Generic dict wasn't possible due to locator change from 4.7 and 4.8, but code is modified and is compatible with both 4.7 and 4.8 versions.

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 11]
```diff
@@ -51,9 +52,52 @@ def create_pvc_ui(self, sc_type, pvc_name, access_mode, pvc_size):
         logger.info("Select PVC size")
         self.do_send_keys(self.pvc_loc["pvc_size"], text=pvc_size)
 
+        if (
+            sc_type
+            in ("ocs-storagecluster-ceph-rbd-thick", "ocs-storagecluster-ceph-rbd")
+            and access_mode == "ReadWriteOnce"
+        ):
+            logger.info(
+                "Test running on OCP version" + ":" + str({get_running_ocp_version()})
```
Comment (petr-balogh) [inline ocs_ci/ocs/ui/pvc_ui.py:61]:
```suggestion
                f"Test running on OCP version: {get_running_ocp_version()}"
```
Didn't get your origin formatting.

File: ocs_ci/ocs/resources/pvc.py [hunk 12]
```diff
@@ -115,6 +115,16 @@ def backed_sc(self):
         """
         return self.data.get("spec").get("storageClassName")
 
+    @property
+    def get_pvc_vol_mode(self):
+        """
+        Function to get pvc volume_mode
+
+        Returns:
+            (str): The volumeMode Value of pvc_obj
```
Comment (petr-balogh) [inline ocs_ci/ocs/resources/pvc.py:124]:
```suggestion
            str: The volumeMode Value of pvc_obj
```
there should be one empty line after last section

File: tests/ui/test_pvc_ui.py [hunk 13]
```diff
@@ -20,25 +21,97 @@ def teardown(self):
...
             pytest.param(
-                *["ocs-storagecluster-cephfs", "test-pvc-fs", "ReadWriteMany", "2"]
-            )
+                *[
+                    "ocs-storagecluster-cephfs",
+                    "test-pvc-fs",
+                    "ReadWriteMany",
+                    "2",
+                    "Filesystem",
+                ]
```
Comment (petr-balogh) [inline tests/ui/test_pvc_ui.py:36]:
```suggestion
                 "ocs-storagecluster-cephfs",
                 "test-pvc-fs",
                 "ReadWriteMany",
                 "2",
                 "Filesystem",
```
Should not this be written only like this? Instead of expanding list which is one extra step here? It also using some space in memory when you define this list which is not used anywhere else.

Or what's the point here?

File: tests/ui/test_pvc_ui.py [hunk 14]
```diff
@@ -20,25 +21,97 @@ def teardown(self):
...
+            ),
+            pytest.param(
+                *[
```
Comment (petr-balogh) [inline tests/ui/test_pvc_ui.py:39]: the same for the rest

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 11 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: updated with f string as suggested

File: tests/ui/test_pvc_ui.py (hunk 13 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: valid point, changed as suggested

File: tests/ui/test_pvc_ui.py (hunk 14 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: done

File: ocs_ci/ocs/resources/pvc.py (hunk 12 above)
Comment (am-agrawa) [reply ocs_ci/ocs/resources/pvc.py]: done

File: ocs_ci/ocs/ui/views.py [hunk 15]
```diff
@@ -240,10 +244,26 @@
...
+    "test-pvc-rbd": ("a[title='test-pvc-rbd']", By.CSS_SELECTOR),
+    "Block": ("input[value='Block']", By.CSS_SELECTOR),
+    "Filesystem": ("input[value='Filesystem']", By.CSS_SELECTOR),
+}
+
+pvc_4_8 = {
+    "ReadWriteMany": ("input[value='ReadWriteMany']", By.CSS_SELECTOR),
+    "pvc_actions": ("button[aria-label='Actions']", By.CSS_SELECTOR),
+    "ReadWriteOnce": ("input[value='ReadWriteOnce']", By.CSS_SELECTOR),
+    "Filesystem": ("input[value='Filesystem']", By.CSS_SELECTOR),
```
Comment (ebondare) [inline ocs_ci/ocs/ui/views.py:261]: Why duplicate the values that are identical in 4.7 and 4.8?

Comment (am-agrawa) [reply ocs_ci/ocs/ui/views.py]: Removed identical locators, please close the requested change so that we can merge the PR.

**Review by petr-balogh — APPROVED** [review]

File: ocs_ci/ocs/ui/views.py [hunk 16]
```diff
@@ -240,10 +244,23 @@
         By.CSS_SELECTOR,
     ),
     "confirm_pvc_deletion": ('button[data-test="confirm-action"]', By.CSS_SELECTOR),
-    "pvc_test": ('a[data-test-id="test-pvc-fs"]', By.CSS_SELECTOR),
     "search_pvc": ('input[data-test-id="item-filter"]', By.CSS_SELECTOR),
 }
 
+pvc_4_7 = {
+    "test-pvc-fs": ('a[data-test-id="test-pvc-fs"]', By.CSS_SELECTOR),
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/views.py:251]: @am-agrawa Is there `data-test-id` attribute for `test-pvc-fs` element?

Comment (am-agrawa) [reply ocs_ci/ocs/ui/views.py:251]: It was added by you for OCP 4.7 and has changed in 4.8.

**Review by OdedViner — APPROVED** [review]

**Review by ebondare — APPROVED** [review]
